tdms/tests/unit/test_utils.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file test_utils.cpp
+ * @brief Tests of the utility functions
+ */
+#include "utils.h"
+
+#include <fstream>
+#include <stdexcept>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "unit_test_utils.h"
+
+TEST_CASE("Test nonexistant file") {
+  REQUIRE_THROWS_AS(assert_can_open_file("DOES_NOT_EXIST.mat", "r"),
+                    std::runtime_error);
+}
+
+TEST_CASE("Test existing file") {
+
+  // create a oneline text file
+  auto tmp = tdms_tests::create_tmp_dir();
+  auto path = tmp / "DOES_EXIST.txt";
+  std::ofstream dummyfile(path.c_str());
+  dummyfile << "Hello" << std::endl;
+  dummyfile.close();
+
+  // should be able to open it
+  REQUIRE_NOTHROW(assert_can_open_file(path.c_str(), "r"));
+  REQUIRE_NOTHROW(assert_can_open_file(path.c_str(), "a"));
+  REQUIRE_NOTHROW(assert_can_open_file(path.c_str(), "a+"));
+
+  REQUIRE_THROWS_AS(assert_can_open_file(path.c_str(), "NONEXISTANT_FILE_MODE"),
+                    std::runtime_error);
```

Comment:
One could argue that this is pointless, I'm using an `std::stringstream` to test a function tha basically wraps `fopen`. 🙃